### ISSUE TITLE
Sequel Pro has poor file handling support

### ISSUE THREAD

**Issue by donblaylock** [sequelpro/sequelpro#2106]
In query view, open a sql file.  open another tab then in query view open another sql file. now click on the first tab and save your sql file.  HOLY CRAP! you just overwrote the second file.  Closing the tab should also prompt you to save any unsaved changes.
Sequel Pro should keep track of the file in each tab and when you click Save it should have the filename opened in that tab not the last opened file.
I love Sequel Pro EXCEPT for this MAJOR flaw.  PLEASE, PLEASE, PLEASE FIX THIS!!!!!
I have accidentally overwritten several files because of this MAJOR BUG!

**Comment by dmoagx** [comment]
> In query view, open a sql file. open another tab then in query view open another sql file. now click on the first tab and save your sql file. HOLY CRAP! you just overwrote the second file. 

This behavior has been changed in development already.
Also I didn't see Sequel Pro "just overwriting" files. Sequel Pro always asked which filename you wanted to use upon save. It just didn't show a warning when filename was already used.

> Closing the tab should also prompt you to save any unsaved changes.

The editor is a scratch area for SQL. It is not meant to be used as a proper text file editor - at least for now.
The "Open"/"Save" metaphor for SQL files makes this confusing for most users. I personally see it as "Copy to Editor / Copy to File"

**Comment by donblaylock** [comment]
Max,

Thank you for replying to my rants.  :o)
I use this tool all day every day and would pay for a “PRO” version if it handled files better.  I will sometimes have 10 or more SQL files open at one time and it gets a little confusing which file I am saving.  I have started putting the filename at the top of my SQL files.  It would be so much nicer if it would keep track of the file that is open in a particular tab and save to that file when you hit save rather than the last opened file.
I bought Navicat Pro in hopes that it would serve me better but it keeps blowing up on my Mac.
If I had time I would download the source code for your app and fix it myself.  But I’m swamped with making a living.  :o)

Looking forward to the next release.

Don Blaylock
DonBlaylock@me.com
406-257-1865 home
406-212-7876 cell

On May 3, 2015, at 12:36 PM, Max Lohrmann <notifications@github.com> wrote:

In query view, open a sql file. open another tab then in query view open another sql file. now click on the first tab and save your sql file. HOLY CRAP! you just overwrote the second file.

This behavior has been changed in development already.
Also I didn't see Sequel Pro "just overwriting" files. Sequel Pro always asked which filename you wanted to use upon save. It just didn't show a warning when filename was already used.

Closing the tab should also prompt you to save any unsaved changes.

The editor is a scratch area for SQL. It is not meant to be used as a proper text file editor - at least for now.
The "Open"/"Save" metaphor for SQL files makes this confusing for most users. I personally see it as "Copy to Editor / Copy to File"
